### ISSUE TITLE
Projen npm package not updated with generated package

### ISSUE THREAD

**Issue by automartin5000** [projen/projen#3471]
The problem of the underlying JSII Node runtime being out of sync with the JSII-compiled package isn't new to me coming from the CDK. However, unless I'm mistaken, this seems to be the case here as well. So in order to update projen and/or custom projects types, a user would have to update both the underlying npm package as well as their runtime's package. Is that correct? If so, I'm surprised keeping these in sync isn't handled automatically in the default task.

**Comment by mrgrain** [comment]
Thanks for the question. Unfortunately I'm not sure I am familiar your scenario. Could you provide me with a more concrete example?

**Comment by automartin5000** [comment]
> Thanks for the question. Unfortunately I'm not sure I am familiar your scenario. Could you provide me with a more concrete example?

Specific example. 
I have a custom Python project that was previously installed with 0.79.xx. I wanted to bump the projen version to get [this update](https://github.com/projen/projen/releases/tag/v0.80.5). However, simply bumping the Python projen package did nothing. It wasn't until I updated the underlying npm package, that the updated env var showed up.

In hindsight, it's possible I didn't run `npx projen` again after bumping the python dep, so it might have been coincidence that running `npm install projen@0.80.12` fixed the problem. But I do know I've seen similar behavior with the AWS CDK.

**Comment by mrgrain** [comment]
Thanks, now I'm with you. That behavior seems odd to me. Are you using a `.projenrc.py` file or a TS file?

**Comment by automartin5000** [comment]
> Thanks, now I'm with you. That behavior seems odd to me. Are you using a `.projenrc.py` file or a TS file?

It's a `.projenrc.py`.

But I think it may have been that I got confused when not running `npx projen` a second time after updating, because I believe the node modules are packaged with the Python package, correct? That said, the relationship between the projen package installed in package.json and the one bundled with the Python package isn't clear to me.

**Comment by mrgrain** [comment]
Yes, the python package and the Node.js package are different things. I'm not actively using projen with python, so my personal experience is limited. But it strikes me odd that you have a `package.json` at all! Do you know why you have that?

Personally I'd use the python binary to run projen.

**Comment by automartin5000** [comment]
> Yes, the python package and the Node.js package are independent of each other. Your `.projenrc.py` file will be using the python package.
> 
> I'm not actively using projen with python, so my personal experience is limited. But it strikes me odd that you have a `package.json` at all! Do you know why you have that?
> 
> Personally I'd use the python binary to run projen for a python package.

Interesting. The Node Projen package is installed when I first create a project with `npx projen new`. Wouldn't that be the only way to do it?

**Comment by mrgrain** [comment]
Hmm. I'll need to check this, it doesn't feel like intentional behaviour but I'm not sure. The alternative is to run `pipx run projen new`. This isn't well documented at the moment, but fully supported.

**Comment by automartin5000** [comment]
`pipx run projen new` seems to have the same effect as `npx projen new` for better or worse. I still end up with a `package.json` that has both projen and my custom projen library as dependencies

**Comment by mrgrain** [comment]
I cannot reproduce this with either `pipx` or `npx projen`:

```console
$ pipx run projen new python --poetry
$ cat package.json
cat: package.json: No such file or directory
```

```console
$ npx projen new python --poetry
$ cat package.json
cat: package.json: No such file or directory
```

**Comment by automartin5000** [comment]
> I cannot reproduce this with either `pipx` or `npx projen`:
> 
> ```
> $ pipx run projen new python --poetry
> $ cat package.json
> cat: package.json: No such file or directory
> ```
> 
>  ```
> $ npx projen new python --poetry
> $ cat package.json
> cat: package.json: No such file or directory
> ```

That's interesting. The only place I see that I'm explicitly defining the dependency is in my library's root projenrc as a "peerDep". Is that the correct configuration? Could that be adding the npm dep?

**Comment by mrgrain** [comment]
I can't imagine that's the cause. `package.json` is only created by a `NodeProject` to my knowledge, or specifically by the `NodePackage` component.

How does the `package.json` file look? Did you maybe run `npm install projen` at some point?

Is `package.json` listed in `.projen/files.json`?
